### ISSUE TITLE
@typescript-eslint/ban-types object not being handled.

### ISSUE THREAD

**Issue by goldingdamien** [typescript-eslint/typescript-eslint#1135]
Config is basically the same as this example but using "object" as well as other examples.
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/ban-types.md

```TS
function test(a: object) {
    console.log(a)
}

function test1(a: Object) {
    console.log(a)
}
```

**Expected Result**
Both object and Object show an error.
I'd actually expect any word regardless of case, defaults or anything else that may be the cause to work.

**Actual Result**
Don't use 'Object' as a type. Use Record<string, any> instead  @typescript-eslint/ban-types

**Additional Info**
No errors. Debug output shows nothing relevant.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.0.0` |
| `@typescript-eslint/parser`        | `2.0.0` |
| `TypeScript`                       | `3.5.3` |
| `ESLint`                           | `6.1.0 |
| `node`                             | `12.1.0 |
| `npm`                              | `6.4.1` |

**Comment by bradzacher** [comment]
See #848

The default configuration does not include handling for `object`.
If you want the rule to ban `object`, you must configure it yourself for now.

---

If you don't like the default suggestion for `Object`, you must configure it yourself for now.
The default configuration does include handling for `Object`, though when I added it, I didn't fully understand the scope of the `Object` type, so I completely misrepresented it.

**Comment by goldingdamien** [comment]
Sorry, what do you mean by "configure it yourself"?
If I have a rule like below and only object is not showing an error, what should I do to get object to show an error?
```
 "@typescript-eslint/ban-types": ["error", {
            "types": {
                "a": null,
                "A": null,
                "object": null,
                "Object": null
             }
       }
]
```

Regards.

**Comment by bradzacher** [comment]
Example, the default config:
https://github.com/typescript-eslint/typescript-eslint/blob/1508670d511052244bf99ac68945064194e4294e/packages/eslint-plugin/src/rules/ban-types.ts#L84-L107

The docs you linked provide other examples of valid config:
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/ban-types.md#options

```
 "@typescript-eslint/ban-types": ["error", {
            "types": {
                "a": null,
                "A": null,
                "object": "do not use object",
                "Object": { 
                  message: 'Use Record<string, any> instead', 
                  fixWith: 'Record<string, any>',
               }
             }
       }
]
```
